### PR TITLE
Use context manager for file handling in misc files.

### PR DESCRIPTION
**Pull request by atombrella**
##### SUMMARY
Inspiration from #63840. No semantic changes. The files were left open.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
miscellaneous.

### CODE REVIEW

File: lib/ansible/module_utils/remote_management/intersight.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def get_gmt_date():
     return formatdate(timeval=None, localtime=False, usegmt=True)
 
 
-class IntersightModule():
+class IntersightModule(object):
```
Comment (atombrella) [inline lib/ansible/module_utils/remote_management/intersight.py]: Probably should be in a different PR. There are other cases of this.

**Review by tcraxs — APPROVED** [review]
LGTM

**Review by felixfontein — APPROVED** [review]
Looks good (and safe) to me. Maybe remove that `object` change though.

**Comment by felixfontein** [comment]
bot_status

(to see which files require core review)

**Comment by felixfontein** [comment]
CC @jillr @mattclay for test/legacy/cleanup_ec2.py

**Review by Andersson007 — COMMENTED** [review]
In postgresql part, LGTM, thanks!

**Comment by atombrella** [comment]
@felixfontein I've remove the `object` addition.

**Review by felixfontein — COMMENTED** [review]
Looks good. Can you add a changelog fragment?

**Comment by atombrella** [comment]
@felixfontein I added one. I think I saw a comment about making one line for each change, instead of one generic. Can you provide feedback for how it should be?

File: changelogs/fragments/65372-misc-context-manager.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+bugfixes:
+  - spacewalk inventory - improve file handling by using a context manager.
+  - intersight remote management - improve file handling by using a context manager.
```
Comment (felixfontein) [inline changelogs/fragments/65372-misc-context-manager.yml]: There are only two intersight modules, so it's best to list them up:
```suggestion
  - intersight_rest_api, intersight_info - improve file handling by using a context manager.
```

File: changelogs/fragments/65372-misc-context-manager.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+  - opennebula - improve file handling by using a context manager.
```
Comment (felixfontein) [inline changelogs/fragments/65372-misc-context-manager.yml]: The module is called one_vm:
```suggestion
  - one_vm - improve file handling by using a context manager.
```

**Review by felixfontein — COMMENTED** [review]
I would name each affected module as concretely as possible:

**Comment by Andersson007** [comment]
```changelogs/fragments/65372-misc-context-manager.yml:6:1: empty-lines: too many blank lines (1 > 0)```

**Comment by atombrella** [comment]
@Andersson007 No worries :) @felixfontein I updated those two module names in the changelog fragment.

**Comment by atombrella** [comment]
I unfortunately closed from my local shell. @felixfontein can you please open it?

**Review by felixfontein — APPROVED** [review]
The changes look safe, and since there was plenty of time to review this, I'm now declaring it done.

**Comment by felixfontein** [comment]
@atombrella thanks for fixing this!
@tcraxs @Andersson007 thanks for reviewing!
